### ISSUE TITLE
CronExpression.parse should recognize Scheduled.CRON_DISABLED 

### ISSUE THREAD

**Issue by larsschaps** [spring-projects/spring-framework#26789]
**Affects:** spring-context:5.3.4

---

Hi...

IMHO 

````Java 
org.springframework.scheduling.support.CronExpression.parse(
   org.springframework.scheduling.annotation.Scheduled.CRON_DISABLED);
````
should not throw java.lang.IllegalArgumentException.

**Comment by poutsma** [comment]
Why? `CronExpression` represents a cron expression, whereas `Scheduled.CRON_DISABLED` represents an annotation value used to disable cron expressions altogether.

**Comment by larsschaps** [comment]
Because the annotation value of `cron` in `org.springframework.scheduling.annotation.Scheduled` a.k.a `@Scheduled` accepts a cron expression as well as `Scheduled.CRON_DISABLED`

So basically `Scheduled.CRON_DISABLED` is a valid cron expression in the context of `@Scheduled`. Not accepting it in `org.springframework.scheduling.support.CronExpression.parse` is inconsistent imho.

**Comment by poutsma** [comment]
>So basically Scheduled.CRON_DISABLED is a valid cron expression in the context of `@Scheduled`.

Indeed, it is a valid value in that context/module, because of way we use annotations to configure scheduling support. But `CronExpression` is a low-level component that knows nothing about annotations. And from the perspective of  `CronExpression`, `CRON_DISABLED` (i.e. `-`) is simply not a valid expression.
